File: mayonez-base/src/main/java/mayonez/graphics/renderer/DrawPriority.java
package mayonez.graphics.renderer;

public enum DrawPriority {

    /**
     * Solid shapes, drawn first.
     */
    FILL(-5),

    /**
     * Shape boundaries, after solid shapes and before lines.
     */
    SHAPE(5),

    /**
     * Lines, drawn after shapes and before points.
     */
    LINE(8),

    /**
     * Single points, drawn last.
     */
    POINT(10);

    public final int zIndex;

    DrawPriority(int zIndex) {
        this.zIndex = zIndex;
    }
}

File: mayonez-base/src/main/kotlin/mayonez/graphics/renderer/DrawPriority.kt
package mayonez.graphics.renderer

enum class DrawPriority(val zIndex: Int) {

    /** Solid shapes, drawn first. */
    SOLID_SHAPE(-5),

    /** Shape outlines, after solid shapes and before lines. */
    SHAPE_OUTLINE(5),

    /** Lines, drawn after shapes and before points. */
    LINE(8),

    /** Single points, drawn last. */
    POINT(10)

}